src/main/java/application/App.java

```diff
@@ -74,13 +74,21 @@ private static Router createRouter() {
         return app;
     }
 
-    private static int setPortNumber(String[] terminal_args) {
-        return terminal_args.length > 0
-                ? Integer.parseInt(terminal_args[0])
-                : 5000;
-    }
+    private static HashMap<String, String> parseServerSettings(String[] args) {
```

Comment:
Why did you decide to implement this in this class? And does this code handle paths that don't exist or port arguments that aren't numbers? I think you're missing validation.

Reply:
If a path does not exist, the server will print out `java.nio.file.NoSuchFileException: /Users/fsadikin/Documents/java_http_server/files` but continue running as when new files are created with this endpoint, the fileHandler will create the necessary base directories to complete the task

You're right -- Validation is needed!.

I put this in the Application class because it feels like it shouldn't be owned by Server or Router, but wasn't sure about creating another Validation/Settings class that the user needs to know about to start the application. What do you think? Would that be a valid reason, or it's ok to have more classes the User should know about?

Reply:
I agree that the arg stuff shouldn't be defined in Server or Router, and for the same reason, I don't think it should be  defined here. I think that the argument parsing is complex enough to deserve its own class. When you say you're concerned about having more classes for a user to know about, are you thinking from the perspective of a user having to write the entire `App` class by themselves? If so, consider that the way you've written it, the user would have to write the arg parsing/validation themselves, which is more work, right?

Reply:
You're right! -- I've created a dedicated Settings class that will take care of terminal arguments validation

Reply:
```
Settings.validateSettings(args);
int port = Settings.getPort();
String directory = Settings.getDirectory();
```